AppCenterData/AppCenterData/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -189,6 +189,106 @@ - (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString
+        MSLogWarning([MSData logTag], @"%@", warningMessage);
+
+        // Delete the local document when found to be expired
+        [self deleteWithToken:token documentId:documentId];
+      } else if ([kMSPendingOperationDelete isEqualToString:documentRow[self.pendingOperationColumnIndex]]) {
+
+        // Ignore document, if the pending operation is found to be Delete
+        NSString *warningMessage = [NSString
+            stringWithFormat:
+                @"Local document pending deletion in local storage for partition '%@' and document ID '%@', excluding from the list",
+                token.partition, documentId];
+        MSLogError([MSData logTag], @"%@", warningMessage);
+      } else {
+
+        // Deserialize document.
+        NSString *jsonString = documentRow[self.documentColumnIndex];
+        NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+        long lastUpdatedDateValue = [(NSNumber *)documentRow[self.operationTimeColumnIndex] longValue];
+        NSDate *lastUpdatedDate = lastUpdatedDateValue > 0 ? [NSDate dateWithTimeIntervalSince1970:lastUpdatedDateValue] : nil;
+
+        // If operation is NSNull, change it to nil.
+        NSString *pendingOperation = [self checkForNSNull:documentRow[self.pendingOperationColumnIndex]];
+
+        // If Etag is NSNull, change it to nil.
+        NSString *etag = [self checkForNSNull:documentRow[self.eTagColumnIndex]];
+        MSDocumentWrapper *cachedDocument = [MSDocumentUtils documentWrapperFromDocumentData:jsonData
+                                                                                documentType:documentType
+                                                                                        eTag:etag
+                                                                             lastUpdatedDate:lastUpdatedDate
+                                                                                   partition:token.partition
+                                                                                  documentId:documentId
+                                                                            pendingOperation:pendingOperation
+                                                                             fromDeviceCache:YES];
+        [localListItems addObject:cachedDocument];
+
+        // Push back cached document expiration time.
+        [self updateDocumentWithToken:token
+                currentCachedDocument:cachedDocument
+                    newCachedDocument:cachedDocument
+                     deviceTimeToLive:deviceTimeToLive
+                            operation:cachedDocument.pendingOperation];
+      }
+    }
+  }
+
+  // Return an empty list if no documents were found.
+  if (localListItems.count == 0) {
+    NSString *warningMessage =
+        [NSString stringWithFormat:@"Unable to find any document in local store for partition '%@'", token.partition];
+    MSLogWarning([MSData logTag], @"%@", warningMessage);
```

Comment:
Same here, just use `MSLogWarning` to format the string, first parameter is a string for formatting